### ISSUE TITLE
csi: aws-efs plugin failing with "Volume capability not supported"

### ISSUE THREAD

**Issue by tgross** [hashicorp/nomad#7360]
In our e2e test for aws-efs (as of #7356's PR), allocations that claim the volume are failing during the pre-run hook with the following task events:

```
Recent Events:
Time                       Type           Description
2020-03-16T14:48:56-04:00  Setup Failure  failed to setup alloc: pre-run hook "csi_hook" failed: rpc error: code = InvalidArgument desc = Volume capability not supported
2020-03-16T14:48:56-04:00  Received       Task received by client
```

The volume mount we're using for the read-only use case is as follows. This sort of thing seems to work fine for the EBS plugin so it may be a matter of passing the right flags for plugin registration or volume registration. I want to circle back to this once #7306 and #7338 are done which will make it easier to test correctly.

```hcl
      volume_mount {
        volume      = "test"
        destination = "${NOMAD_ALLOC_DIR}/test"
        read_only   = true
      }
```

**Comment by tgross** [comment]
This turned out to be simple... the EFS plugin doesn't support single-writer/multi-reader so I need to update the jobspec.

**Comment by tgross** [comment]
Closed by #7387
